draft-smyslov-ipsecme-ikev2-reliable-transport.xml

```diff
@@ -1,148 +1,220 @@
+      TCP. This allows deployments requiring a pure post-quantum key exchange
+      to establish keys during the IKE_SA_INIT exchange without concerns about
+      exceeding typical network MTUs.</t>
+    </section>
+
+    <section anchor="mustshouldmay" title="Terminology and Notation">
+      <t>The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+      "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and
+      "OPTIONAL" in this document are to be interpreted as described in BCP 14
+      <xref target="RFC2119"></xref> <xref target="RFC8174"></xref> when, and
+      only when, they appear in all capitals, as shown here.</t>
+    </section>
+
+    <section anchor="protocol" title="Protocol Details">
+      <section title="Discovery of TCP Transport Support for IKEv2">
+        <t>If the initiator supports this extension, is configured to use it,
+        and does not know whether the responder supports IKEv2 over TCP, the
+        initiator starts the IKE_SA_INIT exchange over UDP port 4500, as per
+        <xref target="RFC7296"></xref>. In this case, the initiator includes
+        the SEPARATE_TRANSPORTS notification (&lt;TBA by IANA&gt;) in the
+        IKE_SA_INIT request. This allows the initiator to discover whether the
+        responder supports the use of separate transports for IKE (over TCP)
+        and ESP (over UDP). Using UDP port 4500 ensures that IPsec traffic can
+        traverse NATs and intermediate devices that allow UDP encapsulation.
+        If the responder supports this extension and receives the
+        SEPARATE_TRANSPORTS notification in the IKE_SA_INIT request, it
+        responds with the same notification in the IKE_SA_INIT response. Upon
+        receiving the SEPARATE_TRANSPORTS notification in the response, the
+        initiator MUST switch to TCP port 4500 for subsequent exchanges
+        (IKE_INTERMEDIATE or IKE_AUTH). The responder MUST be prepared to
+        receive these exchanges over TCP.</t>
+      </section>
+
+      <section title="Using TCP Transport for IKEv2 from the Start">
+        <t>Alternatively, the initiator may start IKE_SA_INIT over TCP port
+        4500 directly, as specified in <xref target="RFC9329"></xref>, for
+        example, when large key exchange payloads (with large public keys) are
+        expected. In this case, the initiator includes the SEPARATE_TRANSPORTS
+        notification in the IKE_SA_INIT request to indicate its preference to
+        use separate transports; IKEv2 over TCP and ESP over UDP, provided
+        that UDP is not blocked in the network path. </t>
+
+        <t>If the responder supports this extension, it responds with the
+        SEPARATE_TRANSPORTS notification in the IKE_SA_INIT response. In this
+        case, child SAs are created as per <xref target="RFC7296"></xref>,
+        with ESP sent over UDP if possible, or over TCP as described in <xref
```

Comment:
ESP can also be sent directly over IP (if no NAT is detected). Perhaps s/over UDP/over UDP (or directly over IP)

Reply:
Fixed.